wsdl/ver10/schema/onvif.xsd

```diff
@@ -7977,14 +7981,34 @@ and sample rate. </xs:documentation>
 		</xs:attribute>
 		<xs:anyAttribute processContents="lax"/>
 	</xs:complexType>
+
+	<xs:complexType name="RecordingEventFilter">
+		<xs:sequence>
+			<xs:element name="TopicFilter" type="xs:string">
+				<xs:annotation><xs:documentation>Topic filter as defined in section 9.6.3 of the ONVIF Core Specification.</xs:documentation></xs:annotation>
+			</xs:element>
+			<xs:element name="ContentFilter" type="xs:string" minOccurs="0">
+				<xs:annotation><xs:documentation>Optional message content filter as defined in section 9.4.4 of the ONVIF Core Specification. 
+					When this parameter is absent any message Data parameter with value other than 'false', zero or empty shall result in a match.</xs:documentation></xs:annotation>
+			</xs:element>
+			<xs:element name="Before" type="xs:duration" minOccurs="0">
```

Comment:
It is possible some device vendors can support Before & After values at System Level rather than at each RecordingJobConfiguration. So  a capability parameter is needed to signal ONVIF client to say device supports Before & After values at each RecordingJobConfiguration or not.

Reply:
Adding a capability should be straightforward. How about:
**Global [tt:StringList]** Settings shared between different recordings and jobs. See trc:GlobalParams for options.

enum trc:GlobalParams {Before, After}

The idea is that changing such a parameter can potentially also affect other recordings.
If needed we can add more constraints in the future.

Reply:
Capability flag is needed  for ONVIF client to know whether device support Event Recording parameters i.e Before & After at Global or as part of each Recording Job Configuration.

Yes, Addition of Global params gives flexibility for ONVIF Client to configure Before & After params at Global Level or Individual Recording Job Configuration level. 

What happens when ONVIF client tries to configure Event recording params  as part of Recording Job Configuration, but the device supports event recording params at Global level? is device should throw the error in this case or allow the client configuration so that all recording job configurations in the device are updated with new event recording params from the client.

Reply:
We have two options:

1. Add APIs to get/set global parameters via a new command GetOptions and SetOptions. 

2.  Add notes to CreateRecordingJob and SetRecordingJobConfiguration that in case of Global dependencies changes to the API may affect other recordings on the device.

Option one needs a lot of extension with dependencies on capabilities to make backward compatible while option two works with existing APIs. If only one company has the restriction I tend to option 2, if many have then I'd prefer option one.

Reply:
We are fine with Option 2. Same time device capability flag is needed i.e whether  device can support event recording settings at global or individual recording job level.